src/Guide/Api/Types.hs

```diff
@@ -284,7 +287,14 @@ toCItemFull Item{..} = CItemFull
   , cifNotes       = H $ toCMarkdown _itemNotes
   , cifLink        = H $ _itemLink
   , cifKind        = H $ _itemKind
+  , cifToc         = H $ map treeToCMD (markdownTreeMdTOC _itemNotes)
   }
+  where
+    treeToCMD = fmap
+        (\(nodes, source) -> CMarkdown
```

Comment:
It's not `(nodes, source)`. It's `(nodes, slug)` (where "slug" is the part after `#` in the link). I guess it's be nice to put that into an ADT, because currently the definition of `MarkdownTree` doesn't make it clear at all and there are no comments either.

To get the source as well, you should change `[Node]` to `[MarkdownInline]` in the definition of `MarkdownTree`.

Reply:
> I guess it's be nice to put that into an ADT

This will also give us a nicer schema for free. (Useful for the frontend)

Reply:
NB: the slug is needed on the frontend as well, of course, because otherwise it can't generate a link.

Reply:
> To get the source as well, you should change [Node] to [MarkdownInline] in the definition of MarkdownTree

Doing this probably results in DB migration which is not a pleasant thing to do. I think it should be okay for now to do some calculations on the fly and change type later when we will switch Database.

Reply:
Migrating `MarkdownTree` is not complicated (because you don't need to migrate any types that contain `MarkdownTree`, only the type itself), but sure, we can also do calculations on the fly.

The problem is that you can't get the source from `Node` on the fly, not faithfully anyway. Or at least I don't see how this could be done.

Reply:
@neongreen Could you, please, provide more details on the expected type of the `MarkdownTree` fields and where is the place to perform migration? So we can create smaller subtasks and paralleliese iisues